flight/Modules/Actuator/actuator.c

```diff
@@ -191,76 +214,34 @@ static void actuatorTask(void* parameters)
 	{
 		PIOS_WDG_UpdateFlag(PIOS_WDG_ACTUATOR);
 
-		// Wait until the ActuatorDesired object is updated
-		bool rc = PIOS_Queue_Receive(queue, &ev, FAILSAFE_TIMEOUT_MS);
+		bool rc=false;
 
-		/* Process settings updated events even in timeout case so we always act on the latest settings */
-		if (actuator_settings_updated) {
-			actuator_settings_updated = false;
-			ActuatorSettingsGet (&actuatorSettings);
-			actuator_update_rate_if_changed (&actuatorSettings, false);
+		// If it's our first iteration, fall down to getting the
+		// config and enabling failsafe.
+		if (!first) {
```

Comment:
It's unfortunate and a factoring I prefer to avoid, but seemed preferable to the alternative where settings-related code was duplicated and could be erroneously changed only in one place or another.  As it stands, the if evaluation is very very cheap compared to everything involved in draining a queue (locks, likely task switch).  Like 3 to 5 orders of magnitude cheaper.